plugins/cinema4d/import_cast.pyp

```diff
@@ -0,0 +1,431 @@
+import c4d
+import os
+import array
+import math
+from c4d import plugins, bitmaps, Vector, gui, BaseObject
+import mxutils
+
+resDir = os.path.join(os.path.dirname(__file__), "res")
+mxutils.ImportSymbols(resDir)
+with mxutils.LocalImportPath(resDir):
+    from cast import Cast, CastColor, Model, Animation, Instance, File
+
+__pluginname__ = "Cast"
+
+class CastLoader(plugins.SceneLoaderData):
+    def Identify(self, node, name, probe, size):
+        if "cast" in name[-4:]:
+            return True
+        return False
+
+    def Load(self, node, name, doc, filterflags, error, bt):
+        importCast(doc, node, name)
+        return c4d.FILEERROR_NONE
+
+def importCast(doc, node, path):
+    cast = Cast.load(path)
+
+    instances = []
+
+    for root in cast.Roots():
+        for child in root.ChildrenOfType(Model):
+            importModelNode(doc, node, child, path)
+        for child in root.ChildrenOfType(Animation):
+            importAnimationNode(doc, child, path)
+        for child in root.ChildrenOfType(Instance):
+            instances.append(child)
+
+    if len(instances) > 0:
+        importInstanceNodes(doc, instances, path)
+        
+"""
+We are using Blender unpack_list for convinience
+https://github.com/blender/blender/blob/main/scripts/modules/bpy_extras/io_utils.py#L370
+"""
+def unpack_list(list_of_tuples):
```

Comment:
In most instances I've seen you use this, it's completely unnecessary where you can just iterate over the pairs directly instead of slicing an unpacked list.

You should be able to remove those steps.

Reply:
In the beginning I tried to replicate the Blender addon as much as possible, that's how this happened. Will remove them!